cogs/TwitchAlert.py

```diff
@@ -411,54 +411,60 @@ async def loop_check_live(self):
-                                                                                    args=[channel.id])
+            try:
+                if streams_details.get('type') == "live":
+                    current_username = str.lower(streams_details.get("user_name"))
+                    # print(current_username + " is live")
+                    usernames.remove(current_username)
+
+                    sql_find_message_id = \
+                        "SELECT UserInTwitchAlert.channel_id, message_id, custom_message, default_message " \
+                        "FROM UserInTwitchAlert " \
+                        "JOIN TwitchAlerts TA on UserInTwitchAlert.channel_id = TA.channel_id " \
+                        "JOIN (SELECT extension_id, guild_id FROM GuildExtensions " \
+                        "WHERE extension_id = 'TwitchAlert' " \
+                        "  OR extension_id = 'All') GE on TA.guild_id = GE.guild_id " \
+                        "WHERE twitch_username = ?;"
+
+                    results = self.ta_database_manager.database_manager.db_execute_select(
+                        sql_find_message_id, args=[current_username])
+
+                    new_message_embed = None
+
+                    for result in results:
+                        channel = self.bot.get_channel(id=result[0])
+                        try:
+                            # If no Alert is posted
+                            if result[1] is None:
+                                if new_message_embed is None:
+                                    if result[2] is not None:
+                                        message = result[2]
+                                    else:
+                                        message = result[3]
+
+                                    new_message_embed = await self.create_alert_embed(streams_details, message)
+                                    # with concurrent.futures.ThreadPoolExecutor() as pool3:
+                                    #    new_message = await asyncio.get_event_loop(). \
+                                    #        run_in_executor(pool3, self.create_alert_message, int(result[0]),
+                                    #                        streams_details, message)
+
+                                if new_message_embed is not None:
+                                    new_message = await channel.send(embed=new_message_embed)
+                                    sql_update_message_id = """
+                                    UPDATE UserInTwitchAlert 
+                                    SET message_id = ? 
+                                    WHERE channel_id = ? 
+                                        AND twitch_username = ?"""
+                                    self.ta_database_manager.database_manager.db_execute_commit(
+                                        sql_update_message_id, args=[new_message.id, result[0], current_username])
```

Comment:
this looks very familiar to some code further down, could we use a function here to improve maintainability?

Reply:
Short answer, yes. Long answer, I am busy but plan to reorganise and test loop_check_live and loop_team_check_live when I am next free.